lldb/source/API/SBTarget.cpp

```diff
@@ -1342,7 +1342,9 @@ lldb::SBWatchpoint SBTarget::WatchAddress(lldb::addr_t addr, size_t size,
 
   SBWatchpointOptions options;
   options.SetWatchpointTypeRead(read);
-  options.SetWatchpointTypeWrite(eWatchpointWriteTypeOnModify);
+  if (modify) {
+    options.SetWatchpointTypeWrite(eWatchpointWriteTypeOnModify);
+  }
```

Comment:
Nit: [Don’t use braces on simple single-statement bodies of if/else/loop statements
](https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements)